Issue777 mass flow source with wea bus

JayHuLBL opened the pull request:
Following models have been added and these new updates have passed Travis test:
- [x]  `IBPSA.Fluid.Sources.MassFlowSourceFromOutside_T`: flow source that produces prescribed mass flow with prescribed trace substances, outside temperature and mass fraction.
- [x] `IBPSA.Fluid.Sources.MassFlowSourceFromOutside_h`: flow source that produces prescribed mass flow with prescribed trace substances, outside specific enthalpy and mass fraction.
- [x] `IBPSA.Fluid.Sources.Examples.Outside_WeatherBus`: example model for applying both above models.

mwetter commented on IBPSA/Fluid/Sources/Examples/Outside_WeatherBus.mo:
```diff
@@ -0,0 +1,109 @@
+within IBPSA.Fluid.Sources.Examples;
+model Outside_WeatherBus
+  "Test model for source (sink) with weather bus"
+  import IBPSA;
```
Please remove the `import` statement. Dymola adds them automatically but they are not needed.

mwetter commented on IBPSA/Fluid/Sources/Examples/Outside_WeatherBus.mo:
```diff
@@ -0,0 +1,109 @@
+        "Simulate and plot"),
+    Documentation(info="<html>
+<p>
+This model demonstrates how to connect fluid flow component to a
```
components should be plural

mwetter commented on IBPSA/Fluid/Sources/MassFlowSourceFromOutside_T.mo:
```diff
@@ -0,0 +1,176 @@
+  medium.T = T_in_internal;
+  medium.Xi = X_in_internal[1:Medium.nXi];
+  ports.C_outflow = fill(C_in_internal, nPorts);
+  annotation (defaultComponentName="boundary",
```
Please use 3 letters, such as "bou"

mwetter commented on IBPSA/Fluid/Sources/MassFlowSourceFromOutside_T.mo:
```diff
@@ -0,0 +1,176 @@
+          textString="C")}),
+    Documentation(info="<html>
+<p>
+Models an ideal flow source, with prescribed values of flow rate and trace substances, with temperature and composition from outside:
```
Please use shorter line so they can be viewed on github

mwetter commented on IBPSA/Fluid/Sources/MassFlowSourceFromOutside_T.mo:
```diff
@@ -0,0 +1,176 @@
+<ul>
+<li>
+May 21, 2017, by Jianjun Hu:<br/>
+Changed the speficication of temperature and composition inputs with outside condition. Weather bus is used.
```
typo in "speficication", and please add a link to the issue

mwetter commented on IBPSA/Fluid/Sources/MassFlowSourceFromOutside_h.mo:
```diff
@@ -0,0 +1,202 @@
+  connect(weaBus.relHum, relHum);
+  TDryBul_degC = TDryBul - 273.15;
+  p_w = relHum * IBPSA.Utilities.Psychrometrics.Functions.saturationPressure(TDryBul);
+  XiDryBul = 0.6219647130774989*p_w/(pAtm-p_w);
```
This is wrong as your equation is for kg/kg dry air, but Xi is in kg/kg total air. Please use the function
`IBPSA.Utilities.Psychrometrics.Functions.X_pW` to compute `XiDryBul` so that we have consistent implementations across the library.

mwetter commented on IBPSA/Fluid/Sources/MassFlowSourceFromOutside_h.mo:
```diff
@@ -0,0 +1,202 @@
+  TDryBul_degC = TDryBul - 273.15;
+  p_w = relHum * IBPSA.Utilities.Psychrometrics.Functions.saturationPressure(TDryBul);
+  XiDryBul = 0.6219647130774989*p_w/(pAtm-p_w);
+  h_in_calculation = 1006*TDryBul_degC + XiDryBul*(2501014.5+1860*TDryBul_degC);
```
Use the enthalpy calculation from the `Medium` instance -- otherwise, if someone uses a different model for air, or uses a model that has a different value for the 0 enthalpy, the results will be wrong.

mwetter commented on IBPSA/Fluid/Sources/MassFlowSourceFromOutside_h.mo:
```diff
@@ -0,0 +1,202 @@
+          textString="C")}),
+    Documentation(info="<html>
+<p>
+Models an ideal flow source, with prescribed values of flow rate and trace substances, with temperature and specific enthalpy from outside:
```
use shorter lines

mwetter commented on IBPSA/Fluid/Sources/MassFlowSourceFromOutside_h.mo:
```diff
@@ -0,0 +1,202 @@
+<ul>
+<li>
+May 21, 2017, by Jianjun Hu:<br/>
+Changed the speficication of specific enthalpy and composition inputs with outside condition. Weather bus is used.
```
typo in speficication, and add issue number

JayHuLBL commented:
Following changes have been finished and they have passed Travis: 
- [x] deleted the `import IBPSA`
- [x] used short lines
- [x] corrected typos
- [x] updated specific enthalpy calculation with medium instance
- [x] added issue number in revise record

mwetter commented:
@Mathadon Can you please review this PR and merge? Jianjun implemented it, I reviewed it, but it is better if non-LBL people review the work before it gets merged.

Mathadon commented on IBPSA/Fluid/Sources/MassFlowSourceFromOutside_T.mo:
```diff
@@ -0,0 +1,187 @@
+Added <code>unit</code> and <code>quantity</code> attributes.
+</li>
+<li>
+September 29, 2009, by Michael Wetter:<br/>
```
The old revision history has been copied. Should this not be removed?

mwetter replied:
Yes it should be removed and started with "Initial version"

Mathadon commented on IBPSA/Fluid/Sources/MassFlowSourceFromOutside_h.mo:
```diff
@@ -0,0 +1,204 @@
+Added <code>unit</code> and <code>quantity</code> attributes.
+</li>
+<li>
+September 29, 2009, by Michael Wetter:<br/>
```
old revision history still here

mwetter commented:
@JayHuLBL Since the parameter `use_T_in` was removed, it is indeed better to only have one model. I suggest to rename `MassFlowSourceFromOutside_h` to `MassFlowSource_WeatherData` and delete `MassFlowSourceFromOutside_T`.

mwetter commented:
@mlauster : Can you please review and merge if it is OK.

mlauster commented on IBPSA/Fluid/Sources/MassFlowSource_WeatherData.mo:
```diff
@@ -0,0 +1,189 @@
+<ul>
+<li> Prescribed mass flow rate.</li>
+<li> Boundary composition (trace-substance flow).</li>
+<li> Outside spefici enthalpy.</li>
```
Typo "spefici", should be "specific". I will fix that.
